src/main/java/uk/gov/hmcts/reform/civil/handler/event/BundleCreationTriggerEventHandler.java

```diff
@@ -47,29 +47,31 @@ public class BundleCreationTriggerEventHandler {
     @EventListener
     public void sendBundleCreationTrigger(BundleCreationTriggerEvent event) {
         boolean isBundleCreated = getIsBundleCreatedForHearingDate(event.getCaseId());
-        if (!isBundleCreated) {
-            BundleCreateResponse bundleCreateResponse = bundleCreationService.createBundle(event);
-
-            String caseId = event.getCaseId().toString();
-            StartEventResponse startEventResponse = coreCaseDataService.startUpdate(caseId, CREATE_BUNDLE);
-            CaseData caseData = caseDetailsConverter.toCaseData(startEventResponse.getCaseDetails().getData());
-
-            List<IdValue<Bundle>> caseBundles = caseData.getCaseBundles();
-            caseBundles.addAll(bundleCreateResponse.getData().getCaseBundles()
-                                   .stream().map(bundle -> prepareNewBundle(bundle, caseData)
-                ).collect(Collectors.toList()));
-            CaseDataContent caseContent = prepareCaseContent(caseBundles, startEventResponse);
-            coreCaseDataService.submitUpdate(caseId, caseContent);
-        } else {
+        if (isBundleCreated) {
             log.info("Trial Bundle already exists for case {}", event.getCaseId());
+            return;
         }
+
+        BundleCreateResponse bundleCreateResponse = bundleCreationService.createBundle(event);
+
+        String caseId = event.getCaseId().toString();
+        StartEventResponse startEventResponse = coreCaseDataService.startUpdate(caseId, CREATE_BUNDLE);
+        CaseData caseData = caseDetailsConverter.toCaseData(startEventResponse.getCaseDetails().getData());
+
+        List<IdValue<Bundle>> caseBundles = caseData.getCaseBundles();
+        caseBundles.addAll(bundleCreateResponse.getData().getCaseBundles()
+                               .stream().map(bundle -> prepareNewBundle(bundle, caseData)
+            ).collect(Collectors.toList()));
+        CaseDataContent caseContent = prepareCaseContent(caseBundles, startEventResponse);
+        coreCaseDataService.submitUpdate(caseId, caseContent);
     }
 
     boolean getIsBundleCreatedForHearingDate(Long caseId) {
         boolean isBundleCreated = false;
         CaseData caseData = caseDetailsConverter.toCaseData(coreCaseDataService.getCase(caseId).getData());
         List<IdValue<Bundle>> caseBundles = caseData.getCaseBundles();
-        isBundleCreated = !(caseBundles.stream().filter(bundleIdValue -> bundleIdValue.getValue()
+        isBundleCreated =
+            !(caseBundles.stream().filter(bundleIdValue -> bundleIdValue.getValue().getBundleHearingDate().isPresent()).filter(bundleIdValue -> bundleIdValue.getValue()
```

Comment:
This is different than before.
Before if there was an element in caseBundles then you would have extracted its bundleHearingDate.
Now you filter out the caseBundles that don't have a bundleHearingDate first and then on the remaining ones you run the check.
Right or wrong, this changes the behaviour of the application and we're late for that since everything has already been tested and approved.

I'm willing to sign off on this change as it's fairly small and in theory it won't happen for our bundles but next time we'll handle this situation as a separate bug or tech debt ticket.

Also, the condition is that tests are not altered. If this change resulted in the alteration of a test, then please revert it.